### PR TITLE
PLT-1016 - Add methods for External Workforce management

### PR DESCRIPTION
**Pull request by lb-pno**
This PR adds three methods to manage external workforces (aka BPO organizations) from a project, in particular:

```python
project.add_external_workforce(workforce_id)

project.remove_external_workforce(workforce_id)

project.get_external_workforces()
```

All three methods return a list of `ExternalWorkforce` associated with the project at the end of the operation or an empty list.

### CODE REVIEW

File: libs/labelbox/src/labelbox/schema/project.py [hunk 1]
```diff
@@ -1824,6 +1825,115 @@ def clone(self) -> "Project":
...
+        Raises:
+            LabelboxError: If the external workforce with the given ID cannot be found.
+
+        Note:
+            This method adds an external workforce (organization) to the current project. 
+            The `workforce_id` parameter can be either a string representing the organization ID or an instance of the `ExternalWorkforce` class.
+        """
+        workforce_id = workforce_id.uid if isinstance(workforce_id, ExternalWorkforce) else workforce_id
+
+        mutation = """
```
Comment (adrian-chang) [inline libs/labelbox/src/labelbox/schema/project.py:1903]: Does this require experimental client mode on? Just wondering don't know myself

**Review by adrian-chang — CHANGES_REQUESTED** [review]
See comments

File: libs/labelbox/src/labelbox/schema/project.py [hunk 2]
```diff
@@ -1840,6 +1841,115 @@ def clone(self) -> "Project":
...
+                        name
+                    }
+                }
+            }
+            """
+        
+        result = self.client.execute(mutation, {"projectId": self.uid, "organizationId": workforce_id})
+        
+        if not result:
+            raise LabelboxError(f"Can't find External Workforce {workforce_id}")
```
Comment (sfendell-labelbox) [inline libs/labelbox/src/labelbox/schema/project.py]: Change to `ResourceNotFoundError`

File: libs/labelbox/src/labelbox/schema/project.py [hunk 3]
```diff
@@ -1840,6 +1841,115 @@ def clone(self) -> "Project":
...
+                        name
+                    }
+                    }
+            }
+        """
+
+        result = self.client.execute(mutation, {"projectId": self.uid, "organizationId": workforce_id})
+
+        if not result:
+            raise LabelboxError(f"Can't find External Workforce {workforce_id}")
```
Comment (sfendell-labelbox) [inline libs/labelbox/src/labelbox/schema/project.py]: Change to `ResourceNotFoundError`

File: libs/labelbox/tests/integration/test_external_workforce.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
+from labelbox import Project, ExternalWorkforce
```
Comment (sfendell-labelbox) [inline libs/labelbox/tests/integration/test_external_workforce.py:1]: Remove trailing whitespace from this file (we have some throughout the codebase but prefer to not add more, we really should add this to our lint rules)

File: libs/labelbox/src/labelbox/schema/project.py [hunk 5]
```diff
@@ -42,6 +42,7 @@
 from labelbox.schema.task_queue import TaskQueue
 from labelbox.schema.ontology_kind import (EditorTaskType, OntologyKind)
 from labelbox.schema.project_overview import ProjectOverview, ProjectOverviewDetailed
+from labelbox.schema.external_workforce import ExternalWorkforce
```
Comment (sfendell-labelbox) [inline libs/labelbox/src/labelbox/schema/project.py]: Remove trailing whitespace from this file

Comment (sfendell-labelbox) [reply libs/labelbox/src/labelbox/schema/project.py]: Also, I know this is such a nit, but as long as you're editing it...Could you alphabetize this import to match the rest?

File: libs/labelbox/tests/integration/test_external_workforce.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+import os
+import sys
+
+def _get_workforce_id_for_test() -> str:
+    # Basic function to provide an organization id for the test
+    # org id from https://labelbox.atlassian.net/wiki/spaces/PLT/pages/2110816271/How+to+labelbox-python+SDK+CI+Tests
+    current_version = sys.version_info[:2] 
+
+    if os.environ['LABELBOX_TEST_ENVIRON'] == "staging":
```
Comment (vbrodsky) [inline libs/labelbox/tests/integration/test_external_workforce.py:25]: what is the meaning of this? can you add comments?

**Review by lnash4 — CHANGES_REQUESTED** [review]
so per @manuaero 's request we would not like to use the add and remove endpoints anymore, now that we have built boost requests within the product 

the mutation's name is upsertProjectBoostWorkforce, this will automatically add the alignerr org as an external workforce

then if the project is properly set up, which means they have 

1. ontology configured 
2. data added
3. there are labeling instructions for the ontology

then you can move the project to the 'REQUESTED' status with the 
updateProjectBoostWorkforceStatus mutation

**Comment by lb-pno** [comment]
I am closing the current PR since it is not relevant with Alignerr.

**Pull request closed without merging**